indico/migrations/versions/20230905_1200_e2b69fe5155d_add_user_data_export_table.py

```diff
@@ -0,0 +1,48 @@
+"""Add data_export_requests table
+
+Revision ID: e2b69fe5155d
+Revises: 0c44046dc1be
+Create Date: 2023-05-08 15:37:24.134940
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects.postgresql import ARRAY
+
+from indico.core.db.sqlalchemy import PyIntEnum, UTCDateTime
+from indico.modules.users.models.export import DataExportOptions, DataExportRequestState
+
+
+# revision identifiers, used by Alembic.
+revision = 'e2b69fe5155d'
+down_revision = '0c44046dc1be'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table(
+        'data_export_requests',
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('requested_dt', UTCDateTime, nullable=False),
+        sa.Column('selected_options', ARRAY(sa.Enum(DataExportOptions)), nullable=False),
```

Comment:
I'd prefer to avoid the native enum in favor of our PyIntEnum type (or set `native_enum=False` in case our type doesn't work properly inside arrays which would absolutely not surprise me...)

https://www.postgresql.org/docs/12/sql-altertype.html#id-1.9.3.42.7

From 12+ you can modify enums in transactions, but not being able to use new enum members in the same transactions where they are being added means we'll never be able to do data migrations that add a new enum type, because in case of alembic upgrades all upgrades are done within the same transaction

Reply:
iirc I did this exactly because our enum wasn't compatible. I'll try with `native_enum=False`